scipy/signal/filter_design.py

```diff
@@ -289,20 +290,40 @@ def freqz(b, a=1, worN=None, whole=False, plot=None):
     Returns
     -------
     w : ndarray
-        The normalized frequencies at which `h` was computed, in
+        1D array of normalized frequencies at which `h` was computed, in
         radians/sample.
     h : ndarray
-        The frequency response, as complex numbers.
+        The frequency response, as complex numbers, with shape corresponding
+        to standard broadcasting rules for `a` and `b` along the non-first
+        dimension.
 
     See Also
     --------
+    freqz_zpk
     sosfreqz
 
     Notes
     -----
-    Using Matplotlib's "plot" function as the callable for `plot` produces
-    unexpected results,  this plots the real part of the complex transfer
-    function, not the magnitude.  Try ``lambda w, h: plot(w, abs(h))``.
+    This function operates along the first dimension  of the `a` and `b`
```

Comment:
But it does operate along the first dimension, and does (or should at least) obey broadcasting rules. So in some sense this is a needed documentation update to what was previously just undocumented behavior (operating along axis 0).

Now whether or not it's currently doing what we want is another question. (And your code pasted below goes in that direction.) Should I check what the behavior was for master for multidimensional inputs and replicate it?

Reply:
I doubt the code was originally written with broadcasting in mind.   In 0.19.1, if `b = np.array([[1, -1], [1, 1]])`, both `freqz(b, worN=5)` and `freqz(b, worN=[0.1, 0.2, 0.3, 0.4, 0.5])` result in `ValueError: operands could not be broadcast together with shapes (5,) (2,)`.

I think dealing with this could be part of a future enhancement.  That enhancement should also deal with `worN` being a multidimensional array--it turns out that it seems to work fine in 0.19.1, but that might just be a happy accident of the implementation.  There are no tests for that case.

For now, it seems safest to limit `b` and `a` to scalars or one-dimensional arrays.  We could even be more strict about this, and check that, after passing them through `atleast_1d()`, the dimensions are, in fact, exactly 1, and raise a `ValueError` if they are not.

Reply:
Either way (current PR or your 1d suggestion) we are breaking backward compatibility for the previously unsupported cases. I guess it's more conservative to force 1D, though. I'll make that change.